tools/src/wyvern/tools/typedAST/core/declarations/DefDeclaration.java

```diff
@@ -355,5 +427,12 @@ public static DefDeclaration generateSetter (GenContext ctx, wyvern.tools.typedA
 		return setterDecl;
 		
 	}
-	
+
+    public static StructuralType genericStructuralType() {
+        List<DeclType> bodyDecl = new LinkedList<>(); // these are the declarations internal to the struct
+        bodyDecl.add(new AbstractTypeMember(GENERIC_MEMBER)); // the body contains only a abstract type member representing the generic type
```

Comment:
It will break in non-trivial examples.  I think it's only passing in the current test cases because the typechecking code is not being careful to ensure all nominal types are well-formed.

You can infer that a function is getting a generic parameter passed using the GENERIC_PREFIX+s naming convention mentioned above, if needed.